src/main/java/org/apache/jackrabbit/filevault/maven/packaging/GenerateMetadataMojo.java

```diff
@@ -868,7 +884,53 @@ private Properties computeProperties(String dependenciesString) {
         return fileMap;
     }
 
+    /**
+     * Establishes a session-shareable workDirectory lookup map for the given pluginContext.
+     *
+     * @param pluginContext a Map retrieved from {@link MavenSession#getPluginContext(PluginDescriptor, MavenProject)}.
+     * @return a lookup Map. The key is {@link Artifact#getId()} and value is {@link AbstractPackageMojo#workDirectory}.
+     */
+    @SuppressWarnings("unchecked")
+    static Map<String, File> getArtifactWorkDirectoryLookup(final Map pluginContext) {
+        final String workDirectoryLookupKey = "workDirectoryLookup";
+        if (!pluginContext.containsKey(workDirectoryLookupKey)) {
+            pluginContext.put(workDirectoryLookupKey, new ConcurrentHashMap<String, File>());
+        }
+        return (Map<String, File>) pluginContext.get(workDirectoryLookupKey);
+    }
+
+    /**
+     * Find the other project which produces the provided artifact.
+     *
+     * @param artifact the dependency artifact needle
+     * @return another project that is a dependency of thisProject
+     */
+    MavenProject findModuleForArtifact(final Artifact artifact) {
+        for (MavenProject otherProject : session.getProjects()) {
+            if (otherProject != this.project) {
+                final Artifact otherArtifact = otherProject.getArtifact();
+                if (getModuleArtifactKey(artifact).equals(getModuleArtifactKey(otherArtifact))) {
```

Comment:
the comparison can just use artifact.getId()

Reply:
Actually, I initially tried this. It became much more complicated because the Artifact constructed from a dependency uses `<dependency><type>` in the same segment where an artifact attached to a `MavenProject` uses the `<project><packaging>` value, so the search never matched. I got to the point of creating a cloned Artifact from the `MavenProject.getArtifact()` with the type replaced with `artifact.getArtifactHandler().getExtension()` when I realized that the layout path does this implicitly and results in much cleaner code.

Reply:
I think this deserves a small javadoc comment on getModuleArtifactKey()